### PR TITLE
[eclipse/xtext#1679]ported more code to java

### PR DESCRIPTION
**Pull request by cdietrich**
[eclipse/xtext#1679]ported more code to java
Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

File: org.eclipse.xtext.ui.tests/xtend-gen/org/eclipse/xtext/ui/tests/xmleditor/XmlContentAssistLexerTests.java [hunk 1]
```diff
@@ -34,22 +35,26 @@
...
-    _builder_1.append("RULE_WS \' \'");
-    _builder_1.newLine();
-    _builder_1.append("RULE_ID \'b\'");
-    _builder_1.newLine();
-    _builder_1.append("RULE_ATTR_EQ \'=\'");
-    _builder_1.newLine();
-    _builder_1.append("RULE_TAG_EMPTY_CLOSE \'/>\'");
-    _builder_1.newLine();
-    this.assertLexing(_builder, _builder_1);
+    try {
```
Comment (nbhusare) [inline org.eclipse.xtext.ui.tests/xtend-gen/org/eclipse/xtext/ui/tests/xmleditor/XmlContentAssistLexerTests.java:38]: I wonder if the try-catch block is needed here.

File: org.eclipse.xtext.xbase.ui.tests/src/org/eclipse/xtext/xbase/ui/tests/editor/Bug372950Test.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+	@Test
+	public void testProposeStringBuffer_07() throws Exception {
+		shouldPropose("if (true) { if (o as String != 1 && o instanceof StringBuff", StringBuffer.class);
+	}
+
+	protected void shouldPropose(String input, Class<?> proposedType) throws Exception {
+		ContentAssistProcessorTestBuilder tester = newBuilder();
+		ICompletionProposal[] proposals = tester.append(input).computeCompletionProposals();
+		List<String> proposalStrings = tester.toString(proposals);
+		Assert.assertTrue(proposalStrings.toString(), proposalStrings.contains(proposedType.getSimpleName()));
```
Comment (nbhusare) [inline org.eclipse.xtext.xbase.ui.tests/src/org/eclipse/xtext/xbase/ui/tests/editor/Bug372950Test.java:65]: Please remove `Assert.`

File: org.eclipse.xtext.xbase.ui.tests/src/org/eclipse/xtext/xbase/ui/tests/editor/Bug379238Test.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+	private ParseHelper<XConstructorCall> parseHelper;
+
+	@Inject
+	private TestingXbaseHoverProvider hoverProvider;
+
+	@Test
+	public void testXConstructorWithNullConstructorNoException() throws Exception {
+		XConstructorCall constructorCall = parseHelper.parse("new()");
+		XtextBrowserInformationControlInput result = hoverProvider.getHoverInfo(constructorCall);
+		Assert.assertNull(result);
```
Comment (nbhusare) [inline org.eclipse.xtext.xbase.ui.tests/src/org/eclipse/xtext/xbase/ui/tests/editor/Bug379238Test.java:34]: Please remove `Assert.`

File: org.eclipse.xtext.xbase.ui.tests/src/org/eclipse/xtext/xbase/ui/tests/editor/ContentAssistTest.java [hunk 4]
```diff
@@ -29,29 +29,29 @@
  */
 public class ContentAssistTest extends AbstractXbaseContentAssistTest {
 
-	protected static final String PROJECT_NAME = "ContentAssistTestProject";
+	protected static String PROJECT_NAME = "ContentAssistTestProject";
```
Comment (nbhusare) [inline org.eclipse.xtext.xbase.ui.tests/src/org/eclipse/xtext/xbase/ui/tests/editor/ContentAssistTest.java:32]: `final`

File: org.eclipse.xtext.xbase.ui.tests/src/org/eclipse/xtext/xbase/ui/tests/editor/XbaseResourceForEditorInputFactoryTest.java [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+public class XbaseResourceForEditorInputFactoryTest extends AbstractXbaseUITestCase {
+	@Inject
+	private XbaseResourceForEditorInputFactory editorInputFactory;
+
+	@Test
+	public void testValidationIsDisabled_01() throws Exception {
+		IProject project = workspace.getRoot().getProject("simpleProject");
+		project.create(null);
+		project.open(null);
+		IFile file = project.getFile("Hello.xtext");
```
Comment (nbhusare) [inline org.eclipse.xtext.xbase.ui.tests/src/org/eclipse/xtext/xbase/ui/tests/editor/XbaseResourceForEditorInputFactoryTest.java:37]: Duplicate code in this and the following test. You might like to abstract it out into a separate method

File: org.eclipse.xtext.xbase.ui.tests/src/org/eclipse/xtext/xbase/ui/tests/refactoring/ExpressionUtilTest.java [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+		assertInsertionPoint("try true catch(Exception e) new $String()", "new String()");
+		assertInsertionPoint("try true finally new $String()", "new String()");
+	}
+
+	protected void assertExpressionSelected(String modelWithSelectionMarkup, String expectedSelection) throws Exception {
+		String cleanedModel = modelWithSelectionMarkup.replaceAll("\\$", "");
+		XExpression expression = parse(cleanedModel);
+		int selectionOffset = modelWithSelectionMarkup.indexOf("$");
+		int selectionLength = modelWithSelectionMarkup.lastIndexOf("$") - selectionOffset - 1;
+		XExpression selectedExpression = util.findSelectedExpression(((XtextResource) expression.eResource()),
```
Comment (nbhusare) [inline org.eclipse.xtext.xbase.ui.tests/src/org/eclipse/xtext/xbase/ui/tests/refactoring/ExpressionUtilTest.java:198]: `((XtextResource) expression.eResource())` -> `(XtextResource) expression.eResource()`

File: org.eclipse.xtext.xbase.ui.tests/src/org/eclipse/xtext/xbase/ui/tests/refactoring/ExpressionUtilTest.java [hunk 7]
```diff
@@ -0,0 +1,240 @@
...
+		Assert.assertEquals(expectedSelection,
+				cleanedModel.substring(selectedRegion.getOffset(), selectedRegion.getOffset() + selectedRegion.getLength()));
+	}
+
+	protected void assertSiblingExpressionsSelected(String modelWithSelectionMarkup, String expectedSelection) throws Exception {
+		String cleanedModel = modelWithSelectionMarkup.replaceAll("\\$", "");
+		XExpression expression = parse(cleanedModel);
+		int selectionOffset = modelWithSelectionMarkup.indexOf("$");
+		int selectionLength = modelWithSelectionMarkup.lastIndexOf("$") - selectionOffset - 1;
+		List<XExpression> selectedExpressions = util.findSelectedSiblingExpressions(((XtextResource) expression.eResource()),
```
Comment (nbhusare) [inline org.eclipse.xtext.xbase.ui.tests/src/org/eclipse/xtext/xbase/ui/tests/refactoring/ExpressionUtilTest.java:210]: `((XtextResource) expression.eResource())` -> `(XtextResource) expression.eResource()`

File: org.eclipse.xtext.xbase.ui.tests/src/org/eclipse/xtext/xbase/ui/tests/refactoring/ExpressionUtilTest.java [hunk 8]
```diff
@@ -0,0 +1,240 @@
...
+		ITextRegion selectedRegion = IterableExtensions.reduce(regions, (a, b) -> a.merge(b));
+		Assert.assertEquals(expectedSelection,
+				cleanedModel.substring(selectedRegion.getOffset(), selectedRegion.getOffset() + selectedRegion.getLength()));
+	}
+
+	protected void assertInsertionPoint(String modelWithInsertionMarkup, String expectedSuccessor) throws Exception {
+		String cleanedModel = modelWithInsertionMarkup.replaceAll("\\$", "");
+		XExpression expression = parse(cleanedModel);
+		int selectionOffset = modelWithInsertionMarkup.indexOf("$");
+		XExpression selectedExpression = util.findSelectedExpression(((XtextResource) expression.eResource()),
```
Comment (nbhusare) [inline org.eclipse.xtext.xbase.ui.tests/src/org/eclipse/xtext/xbase/ui/tests/refactoring/ExpressionUtilTest.java:222]: `((XtextResource) expression.eResource())` -> `(XtextResource) expression.eResource()`

File: org.eclipse.xtext.xbase.ui.tests/src/org/eclipse/xtext/xbase/ui/tests/refactoring/NewFeatureNameUtilTest.java [hunk 9]
```diff
@@ -0,0 +1,150 @@
...
+	public void testInvalidNameUpperCase() throws Exception {
+		assertInvalidName("CamelCase", RefactoringStatus.ERROR, "lowercase");
+	}
+
+	protected void assertDefaultName(String modelWithSelectionMarkup, String expectedName) throws Exception {
+		String cleanedModel = modelWithSelectionMarkup.replaceAll("\\$", "");
+		XExpression expression = parse(cleanedModel);
+		int selectionOffset = modelWithSelectionMarkup.indexOf("$");
+		int selectionLength = modelWithSelectionMarkup.lastIndexOf("$") - selectionOffset - 1;
+		XExpression selectedExpression = util.findSelectedExpression(((XtextResource) expression.eResource()),
```
Comment (nbhusare) [inline org.eclipse.xtext.xbase.ui.tests/src/org/eclipse/xtext/xbase/ui/tests/refactoring/NewFeatureNameUtilTest.java:127]: `((XtextResource) expression.eResource())` -> `(XtextResource) expression.eResource()`

File: org.eclipse.xtext.ui.tests/xtend-gen/org/eclipse/xtext/ui/tests/xmleditor/XmlContentAssistLexerTests.java (hunk 1 above)
Comment (cdietrich) [reply org.eclipse.xtext.ui.tests/xtend-gen/org/eclipse/xtext/ui/tests/xmleditor/XmlContentAssistLexerTests.java:38]: generated code

**Pull request merged**